client-app/src/examples/contact/mobile/details/DetailsPanelModel.ts

```diff
@@ -39,20 +41,16 @@ export default class DetailsPanelModel extends HoistModel {
                 {name: 'tags'}
             ]
         });
-    }
 
-    override async doLoadAsync() {
-        const id = window.location.pathname.split('/').pop();
-        const contact = this.appModel.contacts.find(it => it.id === id);
-        this.setCurrentRecord(contact);
-    }
-
-    @action
-    setCurrentRecord(contact: PlainObject) {
-        this.currentContact = contact;
-        if (!isNil(contact)) {
-            this.formModel.init(contact);
-        }
+        this.addReaction({
+            track: () => this.currentContact,
```

Comment:
Similarly with initializing the form - we react to an observable getter that is itself derived from AppModel.contacts